programs/castle-vault/src/instructions/init.rs

```diff
@@ -30,6 +26,17 @@ pub struct VaultConfigArg {
     pub strategy_type: StrategyType,
 }
 
+pub trait YieldSourceInitializer<'info> {
+    fn initialize_yield_srouce(&mut self) -> ProgramResult;
+}
+
+pub fn init_yield_src_handler<'info, T: YieldSourceInitializer<'info>>(
```

Comment:
I think for the sake of consistency, we should break these out into their own file: `init_yield_source.rs` and the fn name can just be `handler` then